homeassistant/components/sensor/meteo_france.py

```diff
@@ -0,0 +1,142 @@
+"""
+Support for Meteo France raining forecast.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.meteo_france/
+"""
+
+import logging
+import datetime
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, ATTR_ATTRIBUTION, TEMP_CELSIUS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['meteofrance==0.2.5']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by Meteo-France"
+CONF_POSTAL_CODE = 'postal_code'
+
+STATE_ATTR_FORECAST = '1h rain forecast'
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+SENSOR_TYPES = {
+    'rain_chance': ['Rain chance', '%'],
+    'freeze_chance': ['Freeze chance', '%'],
+    'thunder_chance': ['Thunder chance', '%'],
+    'snow_chance': ['Snow chance', '%'],
+    'weather': ['Weather', None],
+    'wind_speed': ['Wind Speed', 'km/h'],
+    'next_rain': ['Next rain', 'min'],
+    'temperature': ['Temperature', TEMP_CELSIUS],
+    'uv': ['UV', None],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_POSTAL_CODE): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Meteo-France sensor."""
+    postal_code = config[CONF_POSTAL_CODE]
+
+    from meteofrance.client import meteofranceClient, meteofranceError
+
+    try:
+        meteofrance_client = meteofranceClient(postal_code)
+    except meteofranceError as exp:
+        _LOGGER.error(exp)
+        return
+
+    client = MeteoFranceUpdater(meteofrance_client)
+
+    add_entities([MeteoFranceSensor(variable, client)
+                  for variable in config[CONF_MONITORED_CONDITIONS]])
+
+
+class MeteoFranceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, condition, client):
+        """Initialize the sensor."""
+        self._condition = condition
+        self._client = client
+        self._data = client.get_data()
```

Comment:
Why are we only calling this here, one time? Shouldn't this be called on every update?

We should probably move this to `update`.

Reply:
I'm new to Python but here is my understanding : 
As `client.get_data` returns a `dict`, it passes the reference to the client data
When it's updated after the remote fetch/parsing of the package, the sensor `_data` is up to date.

I updated to call `get_data()` after the update to be sure the reference is never broken for whatever reason may happen